pkg/compare/compare.go

```diff
@@ -357,19 +359,34 @@ func findAllRequestedSupportedTypes(supportedTypesWithGroups map[string][]string
 	return typesIncludingGroup, notSupportedTypes
 }
 
-// Run uses the factory to parse file arguments (in case of local mode) or gather all cluster resources matching
-// templates types. For each Resource it finds the matching Resource template and
-// injects, compares, and runs against differ.
-func (o *Options) Run() error {
-	showManagedFields := false
-
+func runDiff(obj diff.Object, streams genericiooptions.IOStreams) (*bytes.Buffer, error) {
 	differ, err := diff.NewDiffer("MERGED", "LIVE")
+	diffOutput := new(bytes.Buffer)
 	if err != nil {
-		return err
+		return diffOutput, err
 	}
 	defer differ.TearDown()
 
-	printer := diff.Printer{}
+	err = differ.Diff(obj, diff.Printer{}, ShowManagedFields)
```

Comment:
Perhaps, it would be useful to provide `ShowManagedFields` as a flag for the user, in case user wants to configure.

Reply:
added the flag :)